waltz-server/src/main/java/com/wepay/waltz/server/internal/Partition.java

```diff
@@ -267,14 +267,17 @@ public void receiveMessage(Message msg, PartitionClient client) throws Partition
     }
 
     /**
-     * Assigns a sequence number for the given client and stores this information in a map.
+     * Register PartitionClient in partitionClientsConnectionInfo map to validate that
+     * server communicates with client on the latest channel.
      * @param client The client that is interested in this partition.
+     * @param generation The generation of this partition that client is updated to.
      */
-    public void setPartitionClient(PartitionClient client) {
-        synchronized (partitionClientSeqNums) {
-            Long currentSeqNum = partitionClientSeqNums.get(client.clientId());
-            if (currentSeqNum == null || currentSeqNum < client.seqNum()) {
-                partitionClientSeqNums.put(client.clientId(), client.seqNum());
+    public void registerLatestPartitionClient(PartitionClient client, int generation) {
```

Comment:
Why are we renaming the method?

Reply:
renamed back. Originally felt like setter might be a little deceiving as passed partitionClient might not get set.